lib/service/Service.ts

```diff
@@ -161,6 +161,29 @@ class Service {
     await this.pool.addOutbound({ host, port }, nodePubKey, retryConnecting, true);
   }
 
+  /*
+   * Opens a payment channel to a specified node, currency and amount.
+   */
+  public openChannel = async (
+    args: { nodePubKey: string, amount: number, currency: string },
+  ) => {
+    const { nodePubKey, amount, currency } = args;
+    argChecks.HAS_NODE_PUB_KEY({ nodePubKey });
+    argChecks.POSITIVE_AMOUNT({ amount });
+    argChecks.VALID_CURRENCY({ currency });
+    try {
+      const peer = this.pool.getPeer(nodePubKey);
+      await this.swapClientManager.openChannel({
+        peer,
+        amount,
+        currency,
+      });
+    } catch (e) {
+      const errorMessage = e.message || 'unknown';
```

Comment:
Maybe log the error here?

Reply:
Turns out that we're already RPC logging on the `GrpcService` layer:
https://github.com/ExchangeUnion/xud/blob/master/lib/grpc/GrpcService.ts#L178